Docs/Syntax.md

```diff
@@ -0,0 +1,182 @@
+# Syntax Specification
+
+## Structure
+
+Curly brace code blocks introduce a new scope and may or may not
+have some form of header. Statements may appear in any scope
+including the root level scope (you are not forced to use a `main`
+function).
+
+## Data Types
+
+| Name                   | Description    | Syntax |
+| ---------------------- | -------------- | ------ |
+| Unsigned byte          | Use            | u8     |
+| Unsigned short         | The            | u16    |
+| Unsigned integer       | Internet       | u32    |
+| Unsigned long          | Use            | u64    |
+| Signed byte            | The            | i8     |
+| Signed short           | Internet       | i16    |
+| Signed integer         | Use            | i32    |
+| Signed long            | The            | i64    |
+| Boolean                | Internet       | bool   |
+| String                 | Use            | str    |
+| Single precision float | The            | f32    |
+| Double precision float | Internet       | f64    |
+
+## Statements
+
+### Comments
+
+Single line comments start with `//`.
+
+Multiline comments start with `/*` and end with `*/`.
+
+### Code Blocks
+
+```
+{
+    // This is a scope and a code block
+}
+```
+
+### Literals
+
+```
+10
+10u8
+15i32
```

Comment:
what if the compiler just uses the smallest type possible? unsigned could be marked by an `u` at the end and if you don't want to the smallest type possible you could still append `("i" | "u") , ("8" | "16" | "32" | "64")`?

Reply:
This is a realy bad idea:
1. its impossible to determin what the smallest number is
2. the ambiguity can be realy bad when it comes to accuracy and bitoverflow

Reply:
Then we should default to i32 (that seems like the default in every language). If we call or assign something that require e.g. u8 the compiler shouldn't require the `u8` at the end of the number